0023-communication-protocol.adoc

```diff
@@ -0,0 +1,388 @@
+Examples include `/ipfs/id/1.0.0/` and `/ipfs/id/push/1.0.0`: https://github.com/libp2p/specs/tree/master/identify
+
+The advantages of using single-message protocols are:
+
+* short-lived:
+Open a substream, send the message, close it again.
+No need to wait for the other party to reply.
+* very focused:
+These protocols are small and focused, avoiding the trap of over-engineering them.
+* simpler client code:
+If there is only one possible message that can be sent, its shape can be hardcoded by the client.
+Software that uses libp2p has to cater for several protocols being active concurrently anyway.
+Using the protocol layer as the level with which messages vary makes many things a lot simpler.
+In other words, no need to have `type` fields inside the message, the protocol already identifies the type.
+* cheap:
+Opening substreams is very cheap compared to network connections.
+While there is still some inefficiency in terms of opening substreams, a more flexible design seems to be desirable over some added latency.
+Plus, https://github.com/libp2p/specs/tree/master/identify[work is underway] to make this more efficient.
+
+=== Communication requirements
+
+In order to design the ideal communication protocol, we have to identify which data we want to transfer and why.
+This section aims to achieve that.
+
+==== Verifying the details of the swap
+
+One of the primary requirements for the communication protocol is for both parties to verify that they talk about the same swap.
+This is crucial in avoiding unnecessary transactions, as long as both parties are willing to actually swap.
+
+==== Sending over protocol-specific data
+
+For HTLC-based protocols like HAN, HErc20 and HALight, Alice generates a secret.
+The communication protocol must provide a way for Alice to share the hash of this secret with Bob.
+We also have to take into account that in the future, we will have swap protocols that depend on other cryptographic elements like signatures.
+
+==== Sharing identities
+
+Atomic swap protocols center around the idea of providing a time window in which the change in ownership of assets is atomic.
+A core aspect of these protocols are identities.
+The communication protocol must provide a way for both parties to share identities with the other party.
+For on-chain (i.e. Layer 1) protocols like HAN and HErc20, this is always a pair of identities (redeem + refund).
+For Layer 2 protocols like HALight, the identity is a single public key: the identifier of the user's lnd instance.
+We have to accommodate for these differences in protocols.
+
+== Recommendation
+
+Based on the research above, this section presents a proposal for a set of libp2p protocols that can be combined to offer the required functionality.
+
+=== Announcing a swap
+
+Protocol-identifier: `/comit/swap/announce/1.0.0`
+
+This protocol includes two messages:
+
+. The announce message
+. The confirmation message
+
+==== Announce-message
```

Comment:
Note: I moved my comments concerning the `announce` protocol to the Discussion Points section because they seemed to fit better there.